hal/halx86/xbox/part_xbox.c

```diff
@@ -118,16 +119,31 @@ HalpXboxDeviceHasXboxPartitioning(IN PDEVICE_OBJECT DeviceObject,
         return STATUS_NO_MEMORY;
     }
 
+    Offset.QuadPart = 0;
+    Status = HalpXboxReadSector(DeviceObject, SectorSize, &Offset, SectorData);
+    if (!NT_SUCCESS(Status))
+    {
+        return Status;
+    }
+
+    HasMBRPartitioning = (*((UCHAR *) SectorData + SectorSize - 2) == 0x55 && *((UCHAR *) SectorData + SectorSize - 1) == 0xAA);
+    if (HasMBRPartitioning)
+    {
+        *HasXboxPartitioning = FALSE;
+        goto free;
+    }
+
     Offset.QuadPart = XBOX_SIGNATURE_SECTOR * SectorSize;
     Status = HalpXboxReadSector(DeviceObject, SectorSize, &Offset, SectorData);
-    if (! NT_SUCCESS(Status))
+    if (!NT_SUCCESS(Status))
     {
         return Status;
     }
 
     DPRINT("Signature 0x%02x 0x%02x 0x%02x 0x%02x\n",
            *((UCHAR *) SectorData), *((UCHAR *) SectorData + 1), *((UCHAR *) SectorData + 2), *((UCHAR *) SectorData + 3));
     *HasXboxPartitioning = (XBOX_SIGNATURE == *((ULONG *) SectorData));
+free:
```

Comment:
Not a good label name since it's a function in the standard library. `Exit` and `Cleanup` are common.

Reply:
Improved.